src/slider/Slider.tsx

```diff
@@ -100,168 +102,205 @@ type SliderState = {
   value: Animated.Value;
 };
 
-class Slider extends React.Component<
-  SliderProps & Partial<ThemeProps<SliderProps>>,
-  SliderState
-> {
-  panResponder: PanResponderInstance;
-  isVertical: boolean;
-
-  constructor(props: any) {
-    super(props);
-    this.state = {
-      containerSize: { width: 0, height: 0 },
-      trackSize: { width: 0, height: 0 },
-      thumbSize: { width: 0, height: 0 },
-      allMeasured: false,
-      value: new Animated.Value(
-        getBoundedValue(props.value, props.maximumValue, props.minimumValue)
-      ),
-    };
-    this.panResponder = PanResponder.create({
-      onStartShouldSetPanResponder: this.handleStartShouldSetPanResponder.bind(
-        this
-      ),
-      onMoveShouldSetPanResponder: this.handleMoveShouldSetPanResponder.bind(
-        this
-      ),
-      onPanResponderGrant: this.handlePanResponderGrant.bind(this),
-      onPanResponderMove: this.handlePanResponderMove.bind(this),
-      onPanResponderRelease: this.handlePanResponderEnd.bind(this),
-      onPanResponderTerminationRequest: this.handlePanResponderRequestEnd.bind(
-        this
-      ),
-      onPanResponderTerminate: this.handlePanResponderEnd.bind(this),
-    });
-    this.isVertical = props.orientation === 'vertical';
-    this.componentDidUpdate(props, true);
-  }
+const Slider: React.FunctionComponent<SliderProps, SliderState> = (props) => {
+  const _previousLeft = useRef(0);
+  const [allMeasured, setAllMeasured] = useState(false);
+  const [containerSize, setContainerSize] = useState({ width: 0, height: 0 });
+  const [trackSize, setTrackSize] = useState({ width: 0, height: 0 });
+  const [thumbSize, setThumbSize] = useState({ width: 0, height: 0 });
+  const containerSizeValue = useRef(containerSize);
```

Comment:
I don't understand the use of all these `useRef`'s. They seem unecessary.

Reply:
@flyingcircle  The containerSize,trackSize and thumbSize values are calculated during the render i.e. in onLayout. The panrresponder gets declared before onLayout functions are called. So the problem with directly using state is that initially, the state will have value height:0,width:0 and  if we use the state values in the panResponder they wont be the updated one and when the state will be updated i.e. updation of containerSize,thumbSize and trackSize it will again result in rerender of the whole component and again the panresponder will have the initial values of these states.Using the ref's solves this problem. I think i explained it poorly!